docs/api-reference/core.md

```diff
@@ -6202,7 +6202,10 @@ KubeletConfigReserved
 <td>
 <em>(Optional)</em>
 <p>SystemReserved is the configuration for resources reserved for system processes not managed by kubernetes (e.g. journald).
-When updating these values, be aware that cgroup resizes may not succeed on active worker nodes. Look for the NodeAllocatableEnforced event to determine if the configuration was applied.</p>
+When updating these values, be aware that cgroup resizes may not succeed on active worker nodes. Look for the NodeAllocatableEnforced event to determine if the configuration was applied.
+Deprecated: Separately configuring resource reservations for system processes is deprecated in Gardener and will be removed soon.
+Please merge existing resource reservations into the kubeReserved field.
+TODO(MichaelEischer): Drop this field after v1.99 has been released.</p>
```

Comment:
`v1.99` is too soon for an end-user API. Many people and their automation procedures are affected by this change which is why we should give them at least **6 months** to adapt.

Alternatively, what we already did in the past, was to forbid setting fields with a certain Kubernetes version (see https://github.com/gardener/gardener/pull/6570) and then removed the field as soon as we dropped support for all previous Kubernetes versions. This procedure takes much longer, but is safer at the same time.

I don't see any reason we should rush this change, and propose to:
1. Deprecate the field today.
2. Forbid setting it with Kubernetes `v1.31` (-> add TODO here).
3. Remove field when Gardener support for `v1.30` is dropped.

WDYT?

Reply:
`systemReserved` ideally shouldn't be used anymore when the `NewWorkerPoolHash` feature flag from https://github.com/gardener/gardener/pull/9865 is enabled. It's not a strong requirement but avoids an unnecessary node roll when later on getting rid of `systemReserved`. So ideally users remove uses of `systemReserved` before enabling `NewWorkerPoolHash`, but removing it later on is still possible at the cost of a node pool roll when changing `kubeReserved`.

I guess that corner case would be acceptable as it only occurs when users modify their shoots. If that is the case, then I'd be fine with tying the field removal to the used Kubernetes version. If we want to avoid that corner case, then `systemReserved` has to be removed earlier on, as it would block `NewWorkerPoolHash` from moving towards Beta/GA.

Reply:
I guess this corner case avoidance it up to each landscape operator and the communication effort they invest to make people switching, before they turn on the feature gate. 
What I'm more concerned about is, that we block user automation because of a incompatible change we roll out too fast.

Reply:
I'd still prefer to deprecate `systemReserved` at a faster pace than waiting for the removal of K8S v1.30 support, that is , I'd prefer a deprecation in several months.

Once `NewWorkerPoolHash` is enabled, it doesn't really make sense anymore to use `systemReserved`. If we drop `systemReserved` in several months, then it would be easier to ensure that nobody still uses `systemReserved` when the `NewWorkerPoolHash` feature gate moves to beta. When tying the field deprecation to K8S versions, then I'd expect that users wait longer with making the necessary changes and therefore are more likely to end up in a situation where shoots still use `systemReserved` while the feature gate is already enabled. Hence my preference for a shorter deprecation period.

Reply:
No strong objection from my side. Can you adjust the comment, please?

Reply:
I've updated the comment to mention a deprecation in Gardener 1.113 (about 7 months from now).

There's now also a remark at the NewWorkerPoolHash feature gate that `systemReserved` should no longer be used when that feature gate is enable.